grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/GrpcMessageMarshaller.java

```diff
@@ -244,7 +256,12 @@ private <T> Message deserializeProto(PrototypeMarshaller<T> marshaller, ByteBuf
                 stream = CodedInputStream.newInstance(buf.nioBuffer());
             }
             try {
-                final Message msg = prototype.getParserForType().parseFrom(stream);
+                final Message msg;
+                if (useMethodMarshaller) {
+                    msg = (Message) marshaller.parse(new ByteBufInputStream(buf));
+                } else {
+                    msg = prototype.getParserForType().parseFrom(stream);
+                }
                 try {
                     stream.checkLastTagWas(0);
```

Comment:
Shouldn't we check whether `byteBufInputStream.available() == 0` or not when `useMethodMarshaller` is true?